File: src/main/java/um/tesoreria/rest/model/Plan.java
/**
 * 
 */
package um.tesoreria.rest.model;

import java.io.Serializable;
import java.time.OffsetDateTime;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import jakarta.persistence.UniqueConstraint;

import com.fasterxml.jackson.annotation.JsonFormat;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

/**
 * @author daniel
 *
 */
@Data
@Entity
@Table(uniqueConstraints = { @UniqueConstraint(columnNames = { "pla_fac_id", "pla_id" }) })
@EqualsAndHashCode(callSuper = false)
@NoArgsConstructor
@AllArgsConstructor
public class Plan extends Auditable implements Serializable {
	/**
	 * 
	 */
	private static final long serialVersionUID = 8141418205847063278L;

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@Column(name = "clave")
	private Long uniqueId;

	@Column(name = "pla_fac_id")
	private Integer facultadId;

	@Column(name = "pla_id")
	private Integer planId;

	@Column(name = "pla_nombre")
	private String nombre = "";

	@Column(name = "pla_fecha")
	@JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ssZ", timezone = "UTC")
	private OffsetDateTime fecha;

	@Column(name = "pla_publicar")
	private Byte publicar = 0;

	public String getPlanKey() {
		return this.facultadId + "." + this.planId;
	}
	
}

File: src/main/java/um/tesoreria/core/kotlin/model/Plan.kt
package um.tesoreria.core.kotlin.model

import com.fasterxml.jackson.annotation.JsonFormat
import jakarta.persistence.*
import java.time.OffsetDateTime

@Entity
@Table(uniqueConstraints = [UniqueConstraint(columnNames = ["pla_fac_id", "pla_id"])])
data class Plan(

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "clave")
    var uniqueId: Long? = null,

    @Column(name = "pla_fac_id")
    var facultadId: Int,

    @Column(name = "pla_id")
    var planId: Int,

    @Column(name = "pla_nombre")
    var nombre: String = "",

    @Column(name = "pla_fecha")
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ssZ", timezone = "UTC")
    var fecha: OffsetDateTime,

    @Column(name = "pla_publicar")
    var publicar: Byte = 0
) : Auditable() {

    fun getPlanKey(): String {
        return "$facultadId.$planId"
    }

}
